### PR TITLE
Add Oracle SQL - ASSOCIATE STATISTICS

### PR DESCRIPTION
**Pull request by Liangda-w**
Relates #10513 .

Changes proposed in this pull request:
- Add Oracle SQL - [ASSOCIATE STATISTICS](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/ASSOCIATE-STATISTICS.html#GUID-BD02BA6A-32A7-4093-A6B6-BAE860C0F834)
- Add corresponding test cases

@wgy8283335

### CODE REVIEW

**Comment by wgy8283335** [comment]
I‘d like to review this merge request.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 [hunk 1]
```diff
@@ -1813,3 +1809,36 @@ validationClauses
 intoClause
     : INTO tableName
     ;
+
+associateStatistics
+    : ASSOCIATE STATISTICS WITH (columnAssociation | functionAssociation) storageTableClause?
+    ;
+
+columnAssociation
+    : COLUMNS tableName DOT_ columnName (COMMA_ tableName DOT_ columnName)* usingStatisticsType
```
Comment (wgy8283335) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:1818]: Where is schemaName?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/segment/impl/index/ExpectedIndex.java [hunk 2]
```diff
@@ -20,11 +20,17 @@
 import lombok.Getter;
 import lombok.Setter;
 import org.apache.shardingsphere.test.sql.parser.parameterized.jaxb.cases.domain.segment.AbstractExpectedIdentifierSQLSegment;
+import org.apache.shardingsphere.test.sql.parser.parameterized.jaxb.cases.domain.segment.impl.table.ExpectedOwner;
```
Comment (wgy8283335) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/segment/impl/index/ExpectedIndex.java:23]: Why modify this class?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/segment/index/IndexAssert.java [hunk 3]
```diff
@@ -34,7 +37,7 @@
 public final class IndexAssert {
     
     /**
-     * Assert actual column segment is correct with expected column.
+     * Assert actual index segment is correct with expected index.
```
Comment (wgy8283335) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/segment/index/IndexAssert.java:40]: Why modify this class

**Review by wgy8283335 — COMMENTED** [review]
Some questions about this merge request have been asked.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 (hunk 1 above)
Comment (Liangda-w) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:1818]: `tableName` is defined as

```
tableName
    : (owner DOT_)? name
    ;
```

File: shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/segment/index/IndexAssert.java (hunk 3 above)
Comment (Liangda-w) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/segment/index/IndexAssert.java:40]: Index can also have schema/owner (from my understanding, these two are identical), so add owner in IndexAssert.

image

File: shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/segment/impl/index/ExpectedIndex.java (hunk 2 above)
Comment (Liangda-w) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/segment/impl/index/ExpectedIndex.java:23]: Add owner

**Review by wgy8283335 — APPROVED** [review]

**Comment by Liangda-w** [comment]
Hi @tristaZero , is it the time to merge this PR?:)

**Comment by tristaZero** [comment]
@Liangda-w Sure, sorry for my late response.

**Pull request merged**